### PR TITLE
[Dashboard Reliability] Optimize the node head CPU usage

### PR DESCRIPTION
**Pull request by rkooo567**
Signed-off-by: SangBin Cho <rkooo567@gmail.com>

## Why are these changes needed?

1. Do not use unnecessary internal KV get to get the agent port. The port never changes because agent & raylet fate share.
2. Avoid using a blocking internal KV get. This can be slow especially when GCS is loaded (I saw it took about 20~30 seconds to reply in the worst case).
3. Instead of pinging each individual raylet blocking (`await stub.GetNodeStats`), we ping all raylet at once (`asyncio.gather(*[stub.GetNodeStats for stub in node_stubs])`).
4. Increase the default poll frequency from 1 -> 5 seconds. It is not necessary to query the whole cluster state that often (maybe we can also adjust it based on number of nodes, but 5 seconds must be good enough to feel it is real time)

Screen Shot 2022-11-04 at 4 35 15 AM

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
My assumption is after this change, the CPU usage of raylet will be nearly 1~2%. But let's verify it after it is merged.

File: dashboard/modules/node/node_head.py
Comment (ericl) [inline dashboard/modules/node/node_head.py:336]:
```suggestion
                        f"Cannot reach the node, {node_id}, after timeout {TIMEOUT}."
```

File: dashboard/modules/node/node_head.py
Comment (ericl) [inline dashboard/modules/node/node_head.py:340]:
```suggestion
                        f"Cannot reach the node {node_id}. "
```

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
LGTM assuming you've validated the profiling after the changes.

**Comment by ericl** [comment]
LGTM assuming you've validated the profiling after the changes --- this should be easy enough to do using the `setup-dev` script and ` ray stop; ray start --head --port=6379 --object-manager-port=8076 --port=9031 --no-monitor` even in a multi-node cluster.

**Comment by rkooo567** [comment]
@ericl do you suggest to use workspace in this case?

**Comment by rkooo567** [comment]
Hmm maybe I can test this without using  the real cluster. Let me try

**Comment by ericl** [comment]
Yes, you should be able to easily test this on a workspace with the setup dev script, even in a multi node cluster.

**Comment by rkooo567** [comment]
Before 
Screen Shot 2022-11-09 at 5 15 57 PM


After
Screen Shot 2022-11-09 at 5 15 08 PM

**Comment by rkooo567** [comment]
Most of overhead is now messageToDict like actor_head.py. This is something we can improve by not using dict, but for nodes, it is probably not as important

Screen Shot 2022-11-09 at 4 57 10 PM

Dashboard responsiveness is really good. The avg CPU usage seems to be about 9%. I will measure before this PR as well

**Pull request merged**
